worker/src/main/java/gov/cms/ab2d/worker/config/WorkerConfig.java

```diff
@@ -0,0 +1,89 @@
+package gov.cms.ab2d.worker.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.DateUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.channel.ExecutorChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.Pollers;
+import org.springframework.integration.jdbc.lock.DefaultLockRepository;
+import org.springframework.integration.jdbc.lock.JdbcLockRegistry;
+import org.springframework.integration.jdbc.lock.LockRepository;
+import org.springframework.integration.support.locks.LockRegistry;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+import javax.sql.DataSource;
+
+/**
+ * Configures Spring Integration.
+ * We are using Spring Integration, because it provides valuable features out of the box, such as
+ * database polling, channels, and distributed lock implementation.
+ */
+@Slf4j
+@Configuration
+@EnableIntegration
+public class WorkerConfig {
+
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    private JobHandler handler;
+
+
+    /**
+     * sample values used for thread count and queue capacity. Review it later.
+     */
+    @Bean
+    public ThreadPoolTaskExecutor threadPoolTaskExecutor() {
+        final ThreadPoolTaskExecutor taskExecutor = new ThreadPoolTaskExecutor();
+        taskExecutor.setCorePoolSize(5);
+        taskExecutor.setMaxPoolSize(10);
+        taskExecutor.setQueueCapacity(50);
```

Comment:
Let's make `0` the default queue capacity. This will cause the executor to select `SynchronousQueue` as the queue implementation. That way the worker will simply reject new jobs when the thread pool is exhausted rather than queuing them up and waiting. More info at https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html?is-external=true

Reply:
changed QueueCapacity to 0